ckanext/emailasusername/templates/user/new_user_form.html

```diff
@@ -2,8 +2,30 @@
 
 <form id="user-register-form" action="" method="post">
   {{ form.errors(error_summary) }}
-  {{ form.input("name", id="field-username", label=_("Username"), placeholder=_("username"), value=data.name, error=errors.name, classes=["control-medium"], is_required=True) }}
-  {{ form.input("fullname", id="field-fullname", label=_("Full Name"), placeholder=_("Joe Bloggs"), value=data.fullname, error=errors.fullname, classes=["control-medium"]) }}
+
+  <div id="username-input-field">
+    {{ form.input("fullname", id="field-fullname", label=_("Full Name"), placeholder=_("Joe Bloggs"), value=data.fullname, error=errors.fullname, classes=["control-medium"], attrs={'data-module': 'slug-preview-target', 'class': 'form-control'}, is_required=True) }}
+    {% set attrs = {'data-module': 'slug-preview-slug', 'data-module-placeholder': '<username>', 'class': 'form-control'} %}
+    {{ form.prepend('name', id='field-username', label=_('Username'), placeholder=_('username'), value=data.name, error=errors.name, attrs=attrs, is_required=true) }}
+  </div>
+  <style>
+    /* change URL: field to Username: */
+    #username-input-field .slug-preview strong {
+        display: none;
+    }
+    #username-input-field .slug-preview:before {
+        content: 'Username:';
+        font-weight: bold;
+    }
```

Comment:
Before:
![image](https://user-images.githubusercontent.com/2634482/135253265-dc4073a0-4ae6-481c-9604-1537ccd779eb.png)
The `URL` label is hard coded into the javascript. So we need to use css to hide it

After:
![image](https://user-images.githubusercontent.com/2634482/135253350-aaa4c52f-49dc-40ac-ae07-4a2cf4381841.png)
Note: we've lost the `*` required asterisk here as it's not possible to edit this through css-only and would require us making changes to the javascript. I think this is a good enough solution for now.